tests/runtime/service_generation/test_data_stream_source.py

```
@@ -104,6 +104,8 @@ def test_pickle_round_trip_primitive():
     round_trip = pickle.loads(pickle.dumps(inst))
     assert round_trip.to_dict() == inst.to_dict()
 
+    validate_data_stream(round_trip, 2, float)
```

These tests were hiding failures, we never actually _read_ the unpickled stream